src/System.Buffers.ReaderWriter/System/Buffers/Reader/NewArray.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Buffers
+{
+    /// <summary>
+    /// Simple ArrayPool that just new(s) up arrays
+    /// </summary>
+    internal sealed class NewArray<T> : ArrayPool<T>
```

Comment:
Why do we need this?

Reply:
Because we're not actualy renting anything. We can't control lifetime without making the performance terrible for the normal case.

The idea is that there is an out here for those that want the **best** performance but still want to handle allocations in the segment straddling case.